src/compiler/scanner.ts

```diff
@@ -1836,7 +1851,7 @@ namespace ts {
                     default:
                         if (isIdentifierStart(ch, languageVersion)) {
                             pos++;
-                            while (pos < end && isIdentifierPart(ch = text.charCodeAt(pos), languageVersion)) pos++;
+                            while (pos < end && isIdentifierPart(ch = codePointAtAndMaybeAdvance(pos), languageVersion)) pos++;

```

Comment:
Rather than introduce a single use function, you could just use a block for the while loop and advance `pos` like you do on [1458](https://github.com/microsoft/TypeScript/pull/32096/files#diff-a15280067dcb3b0929808762f8352e86R1458). As written it seems like there is a remote possibility of half-advancing the position since you advance both inside `codePointAtAndMaybeAdvance` and in the loop with different conditions.

Reply:
Or perhaps we should have something like a `function charSize(ch: number): number` and advance using `pos += charSize(ch)` rather than blindly doing `pos++` like we do.

Reply:
Fair enough. I'll add it.